lsp/server/src/server.ts

```diff
@@ -85,65 +98,37 @@ connection.onInitialized(() => {
     }
 });
 
-// Settings for Mobile LSP
-export interface MobileSettings {
-    maxNumberOfProblems: number; //max number of diagnostics to detect per document.
-}
-
-// The global settings, used when the `workspace/configuration` request is not supported by the client.
-// Please note that this is not the case when using this server with the client provided
-// but could happen with other clients.
-const defaultSettings: MobileSettings = { maxNumberOfProblems: 1000 };
-let globalSettings: MobileSettings = defaultSettings;
+connection.onDidChangeConfiguration((change) => {
+    const changedSetting =
+        diagnosticsSettingSection
+            .split('.')
+            .reduce((parent, key) => parent[key], change.settings);
 
```

Comment:
Can we give an example about how this data transformation works.

Reply:
added comments explaining how it works.

Reply:
Where is the comment?

Reply:
I guess I clicked the "Add single comment" before pushing the change, my bad.  pushed now.